packages/grafana-data/src/transformations/transformers/rowNumberToField.ts

```diff
@@ -0,0 +1,50 @@
+import { map } from 'rxjs/operators';
+
+import { DataFrame, Field, FieldType } from '../../types/dataFrame';
+import { DataTransformerInfo } from '../../types/transformations';
+import { ArrayVector } from '../../vector';
+
+import { DataTransformerID } from './ids';
+
+export interface RowNumberToFieldTransformerOptions {}
+
+export const rowNumberToFieldTransformer: DataTransformerInfo<RowNumberToFieldTransformerOptions> = {
+  id: DataTransformerID.rowNumberToField,
+  name: 'Row number to field',
+  description: 'Add the row number of the data frame as a field',
+  defaultOptions: {},
+
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        if (!Array.isArray(data) || data.length === 0) {
+          return data;
+        }
+
+        return data.map((frame) => ({
+          ...frame,
+          fields: addRowNumberField(frame),
```

Comment:
Maybe:
```suggestion
          fields: [
             {
        name: 'index',
        values: new IndexVector(frame.length),
        type: FieldType.number,
        config: {
          min: 0,
          max: frame.length - 1,
        },
      }
             ...frame.fields,
          ]
```

https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/vector/IndexVector.ts#L29

Reply:
Didn't know about that! Looks great